### DISCUSSION TITLE
Laravel Jetstream Inertia with vue-cli

### DISCUSSION THREAD

**Discussion by Sairahcaz** [inertiajs/inertia#566] [Help (Vue)] [answered]
Hey guys,

I was wondering if that stack could make sense and how to set it up.
I played around a bit and when I hardcode the output of the 2 blade directives of the app.blade.php `@inertia` and `@routes` and put those in my /frontend/public/index.html, like here https://github.com/yyx990803/laravel-vue-cli-3 and applied my devServer proxy it kind to start working.
Potentially I have now the power of the vue-cli: hot-reload, lint-on-save, and production-ready builds etc.
But the setup is sure not done yet, I started with the routes by fetching them like here discribed: https://github.com/tighten/ziggy#spas-or-separate-repos . I also need the div with the data-page object, which is created by the `@inertia` directive. 

I just played around with these things and was wondering if its worth it, to get deeper, to have the confort of the vue-cli, or if inertia is not really intend to work like this...

Any ideas, tipps and thoughts would be great!

**Comment by Sairahcaz** [comment]
Any ideas here? :D

**Reply by bill-dou** [reply]
Laravel mix has hot option. I have played around converting Tailwind to Vuetify with Jetstream. Will post out once done.

**Comment by reinink** [comment, marked as answer]
So, from what I understand, vue-cli is a tool for scaffolding a client-side only Vue application. It has no server-side component. Meaning if you need data, you'll need to connect an API...which is why this setup doesn't make sense with Inertia.js.

Inertia is meant to be used within a server-side monolith application, such as Laravel or Rails, or even a Node based framework.

I think a better way of getting some of the functionality of vue-cli with Inertia is by using [Vite](https://vitejs.dev). People are [already using](https://sebastiandedeyne.com/vite-with-laravel/inertia/) Vite with Inertia.

Hope that helps!
